core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala

```diff
@@ -531,6 +531,38 @@ trait DeltaSQLConfBase {
           |""".stripMargin)
       .booleanConf
       .createWithDefault(true)
+
+  val DELTA_OPTIMIZE_MIN_FILE_SIZE =
+    buildConf("optimize.minFileSize")
+        .internal()
+        .doc(
+          """Files which are smaller than this threshold (in bytes) will be grouped together
+             | and rewritten as larger files by the OPTIMIZE command.
+             |""".stripMargin)
+        .longConf
+        .checkValue(_ >= 0, "minFileSize has to be positive")
+        .createWithDefault(1024 * 1024 * 1024)
+
+  val DELTA_OPTIMIZE_MAX_FILE_SIZE =
+    buildConf("optimize.maxFileSize")
+        .internal()
+        .doc("Target file size produced by the OPTIMIZE command.")
+        .longConf
+        .checkValue(_ >= 0, "maxFileSize has to be positive")
+        .createWithDefault(1024 * 1024 * 1024)
```

Comment:
Is it a problem that the max and min file size are the same by default? If you assume the input and output size during compaction is the same (which obviously isn't exactly the case in reality), you'll never compact a file to above the min file size threshold. I guess they would just get filtered out as bins of 1 file, but not sure if that's suboptimal (pun intended)

Reply:
min_size will avoid compacting already large files(> 1 GB in this case) s in the table.

Reply:
Yeah I'm just saying compacted files will never become "large", so they always be reconsidered for compaction (and probably ignored from there so maybe not a huge deal)